src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard-v3/health-checks/health-checks.component.html

```diff
@@ -0,0 +1,23 @@
+<ng-container *ngTemplateOutlet="logsLink"></ng-container>
+<ul>
+  <li *ngFor="let check of healthData.health.checks">
+    <span [ngStyle]="check.severity | healthColor"
+          [class.health-warn-description]="check.severity === 'HEALTH_WARN'">
+    {{ check.type }}</span>: {{ check.summary.message }} <br>
+    <div *ngIf="check.type === 'CEPHADM_FAILED_DAEMON'">
+      <b>Failed Daemons:</b>
+      <div *ngFor="let failedDaemons of check.detail; let last = last">
+        {{ failedDaemons.message.split('daemon ')[1].split(' on ')[0] }}
+        on {{ failedDaemons.message.split(' on ')[1].split(' is in ')[0] }}
+        {{ !last ? ', ' : '' }}
```

Comment:
Push this logic to ts file

Reply:
Usually we try to prioritize working within the html template.
But I suppose this could be within an `<ng-template>` of it's own.

Reply:
I agree but when things get complicated its better to use JS/TS inside a file. To make the HTML look clean and easy to maintain the logic.

Reply:
We need to add unit-testing to this logic. We know how the output is, so its easy to create a reproducer/mock.

Also we are now assuming the description detail will look something like this but it could change eventually I guess...